exec/java-exec/src/main/java/org/apache/drill/exec/store/pcap/PcapBatchReader.java

```diff
@@ -0,0 +1,323 @@
+    dstMacAddressWriter = rowWriter.scalar("dst_mac_address");
+    dstIPWriter = rowWriter.scalar("dst_ip");
+    srcIPWriter = rowWriter.scalar("src_ip");
+    srcPortWriter = rowWriter.scalar("src_port");
+    dstPortWriter = rowWriter.scalar("dst_port");
+    packetLengthWriter = rowWriter.scalar("packet_length");
+
+    //Writers for TCP Packets
+    tcpSessionWriter = rowWriter.scalar("tcp_session");
+    tcpSequenceWriter = rowWriter.scalar("tcp_sequence");
+    tcpAckNumberWriter = rowWriter.scalar("tcp_ack");
+    tcpFlagsWriter = rowWriter.scalar("tcp_flags");
+    tcpParsedFlagsWriter = rowWriter.scalar("tcp_parsed_flags");
+    tcpNsWriter = rowWriter.scalar("tcp_flags_ns");
+    tcpCwrWriter = rowWriter.scalar("tcp_flags_cwr");
+    tcpEceWriter = rowWriter.scalar("tcp_flags_ece");
+    tcpFlagsEceEcnCapableWriter = rowWriter.scalar("tcp_flags_ece_ecn_capable");
+    tcpFlagsCongestionWriter = rowWriter.scalar("tcp_flags_ece_congestion_experienced");
+
+    tcpUrgWriter = rowWriter.scalar("tcp_flags_urg");
+    tcpAckWriter = rowWriter.scalar("tcp_flags_ack");
+    tcpPshWriter = rowWriter.scalar("tcp_flags_psh");
+    tcpRstWriter = rowWriter.scalar("tcp_flags_rst");
+    tcpSynWriter = rowWriter.scalar("tcp_flags_syn");
+    tcpFinWriter = rowWriter.scalar("tcp_flags_fin");
+
+    dataWriter = rowWriter.scalar("data");
+    isCorruptWriter = rowWriter.scalar("is_corrupt");
+
+    return true;
+  }
+
+  @Override
+  public boolean next() {
+    while (!rowWriter.isFull()) {
+      if (!parseNextPacket(rowWriter)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  @Override
+  public void close() {
+    try {
+      fsStream.close();
+    } catch (IOException e) {
+      throw UserException.
+        dataReadError().addContext("Error closing InputStream: " + e.getMessage()).build(logger);
+    }
+    fsStream = null;
+    this.buffer = null;
+    this.decoder = null;
+  }
+
+  private void openFile(FileSchemaNegotiator negotiator) {
+    String filePath = null;
+    try {
+      filePath = split.getPath().toString();
+      fsStream = negotiator.fileSystem().open(new Path(filePath));
+      decoder = new PacketDecoder(fsStream);
+      buffer = new byte[BUFFER_SIZE + decoder.getMaxLength()];
+      validBytes = fsStream.read(buffer);
+    } catch (IOException io) {
+      throw UserException.dataReadError(io).addContext("File name:", filePath).build(logger);
+    }
+  }
+
+  private boolean parseNextPacket(RowSetLoader rowWriter) {
+    Packet packet = new Packet();
+
+    if (offset >= validBytes) {
+      return false;
+    }
+    if (validBytes - offset < decoder.getMaxLength()) {
+      getNextPacket(rowWriter);
+    }
+
+    if (packet == null) {
+      return false;
+    }
+    int old = offset;
+    offset = decoder.decodePacket(buffer, offset, packet, decoder.getMaxLength(), validBytes);
+    if (offset > validBytes) {
+      logger.error("Invalid packet at offset {}", old);
```

Comment:
So we just log error and proceed?

Reply:
One of the challenges in network analysis is dealing with a PCAP that has some corrupt data.  In general, I'd prefer that the reader continue, mark it as corrupt, and attempt to get as much data out of it as possible.